monitoring/linear_combination.py

```diff
@@ -0,0 +1,100 @@
+from operator import itemgetter
+
+import iris
+from scriptengine.exceptions import (
+    ScriptEngineTaskArgumentInvalidError,
+    ScriptEngineTaskArgumentMissingError,
+    ScriptEngineTaskRunError,
+)
+from scriptengine.tasks.core import Task, timed_runner
+
+import helpers.cubes
+
+
+class LinearCombination(Task):
+
+    _required_arguments = (
+        "src",
+        "dst",
+    )
+
+    def __init__(self, arguments):
+        LinearCombination.check_arguments(arguments)
+        super().__init__(arguments)
+
+    @timed_runner
+    def run(self, context):
+        self.log_info("Create new netCDF file from linear combination of inputs")
+
+        sources = self.getarg("src", context)
+        if not isinstance(sources, list):
+            self.log_error("Invalid 'src' argument, must be a list")
+            raise ScriptEngineTaskArgumentInvalidError
+
+        result = 0  # Initialize sum of scaled cubes
+        for src in sources:
+            self.log_debug(f"Processing source {src}")
+            try:
+                path, varname = itemgetter("path", "varname")(src)
+            except TypeError:
+                self.log_error("The list items of the 'src' argument must be dicts")
+                raise ScriptEngineTaskArgumentInvalidError
+            except KeyError as e:
+                self.log_error(f"Missing key(s) {e} in 'src' list item")
+                raise ScriptEngineTaskArgumentMissingError
+
+            try:
+                factor = float(src.get("factor", 1.0))
+            except (ValueError, TypeError) as e:
+                self.log_error(
+                    f"Unable to convert the 'factor' argument to a number: {e}"
+                )
+                raise ScriptEngineTaskArgumentInvalidError
+
+            summand = helpers.cubes.load_input_cube(path, varname)
+            summand = helpers.cubes.remove_aux_time(summand)
+
+            try:
+                scaled_summand = factor * summand
+            except Exception as e:
```

Comment:
Since this is one of the blocks that is not covered by tests yet: Have you encountered a situation where this actually fails and the cube is a valid cube? Otherwise catching a general exception here seems a bit too cautious.

Reply:
I am not sure. I have been experimenting a lot with cube math in a notebook when implementing this, but I do not remember specifically to have encountered a situation where I couldn't multiply a cube with a scalar. An overflow _could_ happen, but that was not what I had in mind. Probably more that Iris cubes are somewhat complex things and I thought that even a simple operation could potentially lead to errors.
In the favor of simplicity we could skip the exception handling and see if there ever comes up an error in real use cases. It should be easy to catch.